### ISSUE TITLE
A document describing the Prow pipeline

### ISSUE THREAD

**Issue by kazydek** [kyma-project/test-infra#169]
**Description**

As part of the Prow setup, we need a document explaining how the Kyma-related jobs configured in prow will result in a continuous integration of the Kyma project. That document should be supported by diagrams simplifying the understanding.
The document should outline the minimal scope (MVP), not the current situation so that you could have a communication base in the development phase, which can be adjusted whenever there are changes required.

The document should cover the following topics:
- Describe all jobs configured in `config.yaml` that relate to the `kyma-project/kyma` repository.
- When a contributor is making a change on a pull request, the contributor understands which jobs will be triggered in which order.
- When a pull request gets merged to the `master` branch, the contributor needs to understand which jobs will be triggered in which order.

In short, we neeed pipeline description:
   - Release
   - PR
   - Master

**Reasons**

We need to have an overview of how the whole configuration of Prow works together.

**Comment by aszecowka** [comment]
> - When a contributor is making a change on a pull request, the contributor understands which jobs will be triggered in which order.

In addition to that, it will be nice that contributor can easily learn what every job is doing, it's purpose etc.

**Comment by derberg** [comment]
@kazydek didn't you address the scope of this task a bit with this PR https://github.com/kyma-project/test-infra/pull/514/files? especially full presets description?

**Comment by kazydek** [comment]
@derberg, Yes, a bit :) There is already a doc that describes how to create a regular job for a component and another doc that lists all available presets that are used in ProwJobs. However, the purpose of this task is to list and explain all types of jobs we have (presubmit, postsubmit, periodic), along with their purpose, the order in which they are triggered, and their role in the whole CI pipeline.

**Comment by bszwarc** [comment]
After some consultation with the devs and looking at the docs - our current documentation is quite comprehensive. Jobs are covered:
* https://github.com/kyma-project/test-infra/blob/master/docs/prow/release-jobs.md 
* https://github.com/kyma-project/test-infra/blob/master/docs/prow/component-jobs.md 

What we can add/reorganize is
- job types info (all types with what they do, when they are triggered, and based on which parameters. Jobs are executed in random order, so we can also inform the user about that
- required vs optional jobs (based in which parameters) 
- check the documentation for any possible improvements
